src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -572,6 +573,76 @@ private static bool HasApplicableConditionalMethod(OverloadResolutionResult<Meth
             return result;
         }
 
+        private BoundExpression BindDynamicArgToLocalFunction(
+            CSharpSyntaxNode syntax,
+            CSharpSyntaxNode expression,
+            string methodName,
+            BoundMethodGroup methodGroup,
+            DiagnosticBag diagnostics,
+            CSharpSyntaxNode queryClause,
+            MethodGroupResolution resolution)
+        {
+            // Invocations of local functions with dynamic arguments
+            // don't need to be dispatched as dynamic invocations
+            // since they cannot be overloaded.  Instead, we'll just
+            // emit a standard call with dynamic implicit conversions
+            // for any dynamic arguments. The one exception is params
+            // arguments which cannot be targeted by dynamic arguments
+            // because there is an ambiguity between an array target
+            // and the params element target. See
+            // https://github.com/dotnet/roslyn/issues/10708
+            Debug.Assert(resolution.OverloadResolutionResult.Succeeded);
+            var validResult = resolution.OverloadResolutionResult.ValidResult;
+
+            var args = resolution.AnalyzedArguments.Arguments;
+            var localFunction = validResult.Member;
+            var parameters = localFunction.Parameters;
+            var methodResult = validResult.Result;
+
+            bool hasErrors = false;
+
+            for (int i = 0; i < args.Count; ++i)
+            {
+                var arg = args[i];
+                if (arg.HasDynamicType())
+                {
+                    var param = parameters[methodResult.ParameterFromArgument(i)];
+                    if (param.IsParams)
```

Comment:
I believe this is not how overload resolution checks for params. Please duplicate the logic.